dapps/sponsored-transactions/src/App.tsx

```diff
@@ -67,15 +68,16 @@ export function App() {
 					json={sponsoredTx}
 					action={
 						<Button
-							disabled={!currentAccount || loading}
+							disabled={!currentAccount || currentAccount!.chains[0]! == SUI_MAINNET_CHAIN || loading}
 							onClick={async () => {
 								setLoading(true);
 								try {
 									const bytes = await tx!.build({
-										provider,
+										client,
 										onlyTransactionKind: true,
 									});
-									const sponsoredBytes = await sponsorTransaction(currentAccount!.address, bytes);
+									const chain = currentAccount!.chains[0]!.split(':')[1]!;
```

Comment:
These chains don't have any specified order, so I wouldn't do this. I'd just make this always one network since it's a demo anyway

Reply:
So `currentAccount.chains` doesn't mean the current network it's connected to?

Reply:
It's the chains the wallets supports, not the one that's currently selected. The idea is that dapps can have their own "selected network" state different from the wallet

Reply:
ok removed